pkgs/tools/security/nmap/default.nix

```diff
@@ -27,7 +27,14 @@ in stdenv.mkDerivation rec {
     sha256 = "063fg8adx23l4irrh5kn57hsmi1xvjkar4vm4k6g94ppan4hcyw4";
   };
 
-  patches = ./zenmap.patch;
+  patches = [ ./zenmap.patch ]
+    ++ optional stdenv.cc.isClang (
+      # https://github.com/nmap/nmap/pull/1363
```

Comment:
please add a comment about why this patch is needed

Reply:
I added a comment here with the details of what the patch fixes. I don't know why it's only necessary in clang, since the code looks invalid in any c++11 environment. I'm trusting the description of upstream PR.